public/app/features/search/components/SearchResults.test.tsx

```diff
@@ -1,52 +1,56 @@
 import React from 'react';
-import { shallow, mount } from 'enzyme';
+import { render, screen, fireEvent } from '@testing-library/react';
 import { SearchResults, Props } from './SearchResults';
 import { searchResults } from '../testData';
 import { SearchLayout } from '../types';
 
-const setup = (propOverrides?: Partial<Props>, renderMethod = shallow) => {
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+const setup = (propOverrides?: Partial<Props>) => {
   const props: Props = {
-    //@ts-ignore
     results: searchResults,
-    onSelectionChanged: () => {},
     onTagSelected: (name: string) => {},
-    onFolderExpanding: () => {},
-    onToggleSelection: () => {},
+    onToggleSection: () => {},
     editable: false,
     layout: SearchLayout.Folders,
   };
 
   Object.assign(props, propOverrides);
 
-  const wrapper = renderMethod(<SearchResults {...props} />);
-  const instance = wrapper.instance();
-
-  return {
-    wrapper,
-    instance,
-  };
+  render(<SearchResults {...props} />);
 };
 
 describe('SearchResults', () => {
   it('should render result items', () => {
-    const { wrapper } = setup();
-    expect(wrapper.find({ 'aria-label': 'Search section' })).toHaveLength(2);
+    setup();
+    expect(screen.getAllByLabelText('Search section')).toHaveLength(2);
   });
 
   it('should render section items for expanded section', () => {
-    const { wrapper } = setup();
-    expect(wrapper.find({ 'aria-label': 'Search items' }).children()).toHaveLength(2);
+    setup();
+    expect(screen.getAllByLabelText(/collapse folder/i)).toHaveLength(1);
+    expect(screen.getAllByLabelText('Search items')).toHaveLength(1);
+    expect(screen.getAllByLabelText(/dashboard search item/i)).toHaveLength(2);
   });
 
   it('should not render checkboxes for non-editable results', () => {
-    //@ts-ignore
-    const { wrapper } = setup({ editable: false }, mount);
-    expect(wrapper.find({ type: 'checkbox' })).toHaveLength(0);
+    setup({ editable: false });
+    expect(screen.queryAllByRole('checkbox')).toHaveLength(0);
   });
 
-  it('should render checkboxes for non-editable results', () => {
-    //@ts-ignore
-    const { wrapper } = setup({ editable: true }, mount);
-    expect(wrapper.find({ type: 'checkbox' })).toHaveLength(4);
+  it('should render checkboxes for editable results', () => {
+    setup({ editable: true });
+    expect(screen.getAllByRole('checkbox')).toHaveLength(4);
+  });
+
+  it('should collapse folder item list on header click', () => {
+    const mockOnToggleSection = jest.fn();
+    setup({ onToggleSection: mockOnToggleSection });
+
+    fireEvent.click(screen.getByLabelText('Collapse folder 0'));
+    expect(mockOnToggleSection).toHaveBeenCalledTimes(1);
+    expect(mockOnToggleSection).toHaveBeenCalledWith(searchResults.find(result => result.id === 0));
```

Comment:
nitpick: I think we should try to keep our expect statements raw and avoid using any logic/function to retrieve the expected value.